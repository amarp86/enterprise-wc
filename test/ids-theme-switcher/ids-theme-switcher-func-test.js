/**
 * @jest-environment jsdom
 */
import IdsContainer from '../../src/ids-container/ids-container';
import IdsThemeSwitcher from '../../src/ids-theme-switcher/ids-theme-switcher';

describe('IdsThemeSwitcher Component', () => {
  let container;
  let switcher;

  beforeEach(async () => {
    container = new IdsContainer();
    switcher = new IdsThemeSwitcher();
    container.appendChild(switcher);
    document.body.appendChild(container);
  });

  afterEach(async () => {
    document.body.innerHTML = '';
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    const container2 = new IdsContainer();
    const switcher2 = new IdsThemeSwitcher();
    container2.appendChild(switcher2);
    document.body.appendChild(container2);
    container2.remove();
    switcher2.remove();
    expect(document.querySelectorAll('ids-container').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    expect(switcher.shadowRoot.innerHTML).toMatchSnapshot();
  });

  it('handles selected from the ids-popup-menu', () => {
    const event = new CustomEvent('selected', { detail: { elem: { value: 'classic' } } });
    switcher.shadowRoot.querySelector('ids-popup-menu').dispatchEvent(event);

    expect(switcher.version).toEqual('classic');

    event.detail.elem.value = 'contrast';
    switcher.shadowRoot.querySelector('ids-popup-menu').dispatchEvent(event);

    expect(switcher.version).toEqual('classic');
    expect(switcher.mode).toEqual('contrast');
  });

  it('can set mode and then clear it to default', () => {
    switcher.mode = 'dark';
    switcher.mode = '';
    expect(switcher.mode).toEqual('light');
    expect(switcher.getAttribute('mode')).toBeFalsy();
  });

  it('can set mode and then clear it to default', () => {
    switcher.version = 'classic';
    switcher.version = '';
    expect(switcher.version).toEqual('new');
    expect(switcher.getAttribute('version')).toBeFalsy();
  });
});
