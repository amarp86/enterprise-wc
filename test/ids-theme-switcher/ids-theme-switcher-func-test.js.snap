// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsThemeSwitcher Component renders correctly 1`] = `
"<style nonce=\\"0a59a005\\">.ids-theme-switcher { background-color: transparent; }</style><ids-menu-button id=\\"ids-theme-switcher\\" menu=\\"ids-theme-menu\\" mode=\\"light\\" version=\\"new\\">
            <ids-icon slot=\\"icon\\" icon=\\"more\\"></ids-icon>
            <span class=\\"audible\\">Theme Switcher</span>
        </ids-menu-button>
        <ids-popup-menu id=\\"ids-theme-menu\\" target=\\"ids-theme-switcher\\" trigger=\\"click\\" hidden=\\"\\">
          <ids-menu-group>
            <ids-menu-item mode=\\"light\\" version=\\"new\\">
              Theme
              <ids-popup-menu slot=\\"submenu\\" hidden=\\"\\">
                <ids-menu-group select=\\"single\\">
                  <ids-menu-item selected=\\"true\\" value=\\"new\\" mode=\\"light\\" version=\\"new\\">New</ids-menu-item>
                  <ids-menu-item value=\\"classic\\" mode=\\"light\\" version=\\"new\\">Classic</ids-menu-item>
                </ids-menu-group>
              </ids-popup-menu>
            </ids-menu-item>
            <ids-menu-item mode=\\"light\\" version=\\"new\\">
              Mode
              <ids-popup-menu slot=\\"submenu\\" hidden=\\"\\">
                <ids-menu-group select=\\"single\\">
                  <ids-menu-item selected=\\"true\\" value=\\"light\\" mode=\\"light\\" version=\\"new\\">Light</ids-menu-item>
                  <ids-menu-item value=\\"dark\\" mode=\\"light\\" version=\\"new\\">Dark</ids-menu-item>
                  <ids-menu-item value=\\"contrast\\" mode=\\"light\\" version=\\"new\\">High Contrast</ids-menu-item>
                </ids-menu-group>
              </ids-popup-menu>
            </ids-menu-item>
          </ids-menu-group>
        </ids-popup-menu>"
`;
